spring-ai-core/src/main/java/org/springframework/ai/chat/client/observation/ChatClientObservationDocumentation.java

```diff
@@ -0,0 +1,158 @@
+* limitations under the License.
+*/
+package org.springframework.ai.chat.client.observation;
+
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationConvention;
+import io.micrometer.observation.docs.ObservationDocumentation;
+
+/**
+ * @author Christian Tzolov
+ * @since 1.0.0
+ */
+public enum ChatClientObservationDocumentation implements ObservationDocumentation {
+
+	/**
+	 * AI Chat Client observations
+	 */
+	AI_CHAT_CLIENT {
+		@Override
+		public Class<? extends ObservationConvention<? extends Observation.Context>> getDefaultConvention() {
+			return DefaultChatClientObservationConvention.class;
+		}
+
+		@Override
+		public KeyName[] getLowCardinalityKeyNames() {
+			return LowCardinalityKeyNames.values();
+		}
+
+		@Override
+		public KeyName[] getHighCardinalityKeyNames() {
+			return new KeyName[] {};
+		}
+
+	};
+
+	public enum LowCardinalityKeyNames implements KeyName {
+
+		/**
+		 * Spring AI kind.
+		 */
+		SPRING_AI_KIND {
+			@Override
+			public String asString() {
+				return "spring.ai.kind";
+			}
+		},
+
+		/**
+		 * Is the chat model response a stream.
+		 */
+		STREAM {
+			@Override
+			public String asString() {
+				return "chat.client.stream";
+			}
+		},
+
+		/**
+		 * ChatModel response raw status code, or {@code "IO_ERROR"} in case of
+		 * {@code IOException}, or {@code "CLIENT_ERROR"} if no response was received.
+		 */
+		STATUS {
+			@Override
+			public String asString() {
+				return "chat.client.status";
+			}
+		}
+
+	}
+
+	public enum HighCardinalityKeyNames implements KeyName {
+
+		/**
+		 * Enabled tool function names.
+		 */
+		TOOL_FUNCTION_NAMES {
+			@Override
+			public String asString() {
+				return "spring.ai.chat.client.tool.function.names";
+			}
+		},
+		/**
+		 * List of configured chat client advisors.
+		 */
+		CHAT_CLIENT_ADVISOR {
+			@Override
+			public String asString() {
+				return "spring.ai.chat.client.advisors";
+			}
+		},
+		/**
+		 * Map of advisor parameters.
+		 */
+		CHAT_CLIENT_ADVISOR_PARAM {
+			@Override
+			public String asString() {
+				return "spring.ai.chat.client.advisor.params";
+			}
+		},
+		/**
+		 * Chat client user text.
+		 */
+		CHAT_CLIENT_USER_TEXT {
+			@Override
+			public String asString() {
+				return "spring.ai.chat.client.user.text";
+			}
+		},
+		/**
+		 * Chat client user parameters.
+		 */
+		CHAT_CLIENT_USER_PARAM {
+			@Override
+			public String asString() {
+				return "spring.ai.chat.client.user.params";
+			}
+		},
+		/**
+		 * Chat client system text.
+		 */
+		CHAT_CLIENT_SYSTEM_TEXT {
+			@Override
+			public String asString() {
+				return "spring.ai.chat.client.system.text";
+			}
+		},
+		/**
+		 * Chat client system parameters.
+		 */
+		CHAT_CLIENT_SYSTEM_PARAM {
+			@Override
+			public String asString() {
+				return "spring.ai.chat.client.system.params";
+			}
+		};
+
+		@Override
```

Comment:
Is this needed?

Reply:
which one?

Reply:
The `STATUS` enum value (`chat.client.status`)